TESTING.md

```diff
@@ -1,89 +1,483 @@
+    ...
+    ✔  All specs passed!                        371ms        1        1
+    ```
+
+### Useful information
+
+* In most cases, a single test which covers the following topics is enough:
+  * Login/Logout: Checks the UI, app, and DB are working together
+  * Basic functionality test
+
+* Avoid multiple tests for the same block of functionality.
+
+* If the asset exposes an API, Cypress is an excellent option to test this feature!
+
+### Specific acceptance criteria
+
+* [ ] Test file name has the following format: Helm chart name + spec (ex: `wordpress_spec.js`)
+* [ ] No `describe()` blocks should be present
+* [ ] Aim to have an assertion after every command to avoid flakiness, taking advantage of Cypress retry-ability
+* [ ] Test description is a sentence with the following format: Expected result summary, starting with a verb, in third person, no dots at the end of the sentence (ex: `it('checks if admin can edit a site', ()`)
+* [ ] Respect the folder structure recommended by Cypress:
   * [fixtures](https://docs.cypress.io/api/commands/fixture) - for test data
   * [Integration](https://docs.cypress.io/api/commands/fixture) - test scenario
   * [plugins](https://docs.cypress.io/guides/tooling/plugins-guide) - plugin configuration, if applicable
   * [support](https://docs.cypress.io/api/commands/fixture) - reusable behaviours and overrides
   * [cypress.json](https://docs.cypress.io/guides/tooling/plugins-guide) - configuration values you wish to store
+* [ ] DOM selectors should be resilient. See best practices [here](https://docs.cypress.io/guides/references/best-practices#Selecting-Elements)
+* [ ] Unnecessary waiting should be avoided
+* [ ] Apply the following code style:
+  * Use 2 spaces for indentation
+  * Semicolons should be used
+  * Lines over 80 columns long should preferably be devided into sensible chunks
+  * Prefer single-quotes over double ones for strings
+  * Use template literals (\`The value is: ${var}\`) for string interpolation
+
+## Ginkgo
+
+[Ginkgo](https://onsi.github.io/ginkgo/#top) is one of the frameworks used to implement integration tests. Related files should be located under `/.vib/ASSET/ginkgo`. It is the reference tool to use when tests require interaction with the K8s cluster.
+
+In order for VIB to execute Ginkgo tests, the following block of code needs to be defined in the corresponding [VIB pipeline files](#vib-pipeline-files) (`/.vib/ASSET/vib-{verify,publish}.json`).
+
+> Values denoted withing dollar signs (`$$VALUE$$`) should be treated as placeholders
+
+```json
+        {
+          "action_id": "ginkgo",
+          "params": {
+            "resources": {
+              "path": "/.vib/$$ASSET$$/ginkgo"
+            },
+            "params": {
+              "kubeconfig": "{{kubeconfig}}",
+              "namespace": "{{namespace}}",
+              "$$PARAM_3$$": "$$VALUE_3$$",
+              ...
+            }
+          }
+        }
+```
+
```

Comment:
For Ginkgo, a `Useful information` section could give an overview of the most common use cases:

- Log checking for a scratch/k8s native pod
- Deployment of CRDs objects from an operator
- Usage of the `extraDeploys` parameter to simplify ginkgo logic